### DISCUSSION TITLE
List of All Possible Values for Forecast Properties

### DISCUSSION THREAD

**Discussion by thekendog** [weather-gov/api#386] [Q&A]
Is there a list of all possible values for certain variables, such as the short forecast one from this endpoint?

https://api.weather.gov/gridpoints/TOP/31,80/forecast

**Comment by scadergit** [comment]
The short and detailed forecast are derived from algorithms based on NWS directives. There is no simple list of output values, and the output may change without notice from quality improvements.

Will you provide more details on the intended use of a list? Perhaps we can find a substitute or consider options for future enhancements.

**Comment by thekendog** [comment]
I am trying to apply a certain icon based on the short forecast.  I only have 10 basic icons and therefore need to match up all the different conditions to an appropriate icon.

**Comment by scadergit** [comment]
A keyable forecast value is a common request, so I pinned the question. The short answer is that is not possible, but the team is aware of the popularity of this feature. I'll bring it up for discussion again to see if we can move it forward.

**Comment by albertyw** [comment]
I've been slowly compiling a list of conditions in https://github.com/albertyw/codemancer/blob/master/codemancer/js/weather.js#L10-L42.  Conditions are oftentimes preceded by a descriptor, which I strip out when converting it into an icon.

**Comment by jwdinker** [comment]
@scadergit I maybe missing this somewhere, but it would be nice to have quantitive values exposed to derive the different weather states instead of relying on string enums, especially when those values get concatenated into a long string in the short and long description.  The only alternative at this moment is writing regex's to extract values or relying on descriptive string values like @albertyw .

For instance, the documentation says when the precipitation probability is 20 percent, the expression of uncertainty is "slight chance" and equivalent areal modifier is "widely scattered."  Is it not possible to expose / normalize those values where you'd have something like this at the forecast level...

```
{
...
"precipitation_probability":20,
"cloud_coverage":80
}
```

![2020-02-09_19-35-37](https://user-images.githubusercontent.com/2313028/74113683-c95f2700-4b73-11ea-850f-9b059caf339a.png)

![2020-02-09_19-41-48](https://user-images.githubusercontent.com/2313028/74113744-396dad00-4b74-11ea-9182-c91b5cc7ce28.png)


That would make it a breeze with some simple equality checks.  We can already do this for wind speed.

Thanks!

**Comment by scadergit** [comment]
@jwdinker The forecast endpoint was developed two years ago as MVP to replicate the JSON provided by MapClick on the current version of forecast. The raw gridpoint data was exposed knowing this limitation, and there are plans (towards your suggestion) to ehance the forecast endpoint once it gets priority again.

**Comment by jwdinker** [comment]
@scadergit Makes sense.  I appreciate the explanation.

**Comment by masonhfisher** [comment]
> 
> 
> A keyable forecast value is a common request, so I pinned the question. The short answer is that is not possible, but the team is aware of the popularity of this feature. I'll bring it up for discussion again to see if we can move it forward.

Just curious but why is a keyable forecast value not possible?

**Comment by scadergit** [comment]
@MachineThing Sorry, I meant not possible _yet._ As noted just above, the forecast endpoint has not seen updates in some time, and will be improved once priority allows us to do so.

**Comment by masonhfisher** [comment]
> @MachineThing Sorry, I meant not possible _yet._ As noted just above, the forecast endpoint has not seen updates in some time, and will be improved once priority allows us to do so.

Thanks for the quick reply!

**Comment by at1993** [comment]
The full list can be found here under “weather-summary” https://graphical.weather.gov/xml/xml_fields_icon_weather_conditions.php

**Comment by ghost** [comment]
Is there a list for all the possible conditions for the `weather` field for the raw (gridpoint) endpoint?

~~~json
{
  "coverage": "slight_chance",
  "weather": "rain_showers",
  "intensity": "light",
}
~~~

**Comment by StephenClouse** [comment]
There is, actually. We've been working on updating the OpenAPI definition for API that will be included in our next release. This is the working schema for the weather layer values:

```
{
   "description": "A value object representing expected weather phenomena.",
   "type": "object",
   "properties": {
      "coverage": {
         "type": "string",
         "enum": [
            "areas",
            "brief",
            "chance",
            "definite",
            "few",
            "frequent",
            "intermittent",
            "isolated",
            "likely",
            "numerous",
            "occasional",
            "patchy",
            "periods",
            "scattered",
            "slight_chance",
            "widespread"
         ],
         "nullable": true
      },
      "weather": {
         "type": "string",
         "enum": [
            "blowing_dust",
            "blowing_sand",
            "blowing_snow",
            "drizzle",
            "fog",
            "freezing_fog",
            "freezing_drizzle",
            "freezing_rain",
            "freezing_spray",
            "frost",
            "hail",
            "haze",
            "ice_crystals",
            "ice_fog",
            "rain",
            "rain_showers",
            "sleet",
            "smoke",
            "snow",
            "snow_showers",
            "thunderstorms",
            "volcanic_ash",
            "water_spouts"
         ],
         "nullable": true
      },
      "intensity": {
         "type": "string",
         "enum": [
            "light",
            "moderate",
            "heavy"
         ],
         "nullable": true
      },
      "visibility": {
         "$ref": "#/components/schemas/QuantitativeValue"
      },
      "attributes": {
         "type": "array",
         "items": {
            "type": "string",
            "enum": [
               "damaging_wind",
               "dry_thunderstorms",
               "flooding",
               "gusty_wind",
               "heavy_rain",
               "large_hail",
               "small_hail",
               "tornadoes"
            ]
         }
      }
   },
   "required": [
      "coverage",
      "weather",
      "intensity",
      "visibility",
      "attributes"
   ],
   "additionalProperties": false
}
```

Note this is subject to change before release, as I might have missed something.

**Comment by ghost** [comment]
@StephenClouse This is great! Thank you.

**Comment by SyedaKhadija-sk** [comment]
???
